internal/util/volid.go

```diff
@@ -127,9 +127,10 @@ func (ci *CSIIdentifier) DecomposeCSIID(composedCSIID string) (err error) {
 	ci.ClusterID = composedCSIID[10 : 10+clusterIDLength]
 	// additional 1 for '-' separator
 	bytesToProcess -= (clusterIDLength + 1)
-	nextFieldStartIdx := 10 + clusterIDLength + 1
+	nextFieldStartIdx := (10 + clusterIDLength + 1)
 
-	if bytesToProcess < 17 {
+	const minLenToDecode = 17
```

Comment:
can this be calculated, or maybe `len("csi-bla-ABCD0123")` placeholder string?

Reply:
Thanks Niels, I'll get back with the changes.

Reply:
@nixpanic , Please correct me if I am wrong, but since 17 is a fixed value, then shouldn't computing it every time include an extra step?

Reply:
if you are worried that computing is problematic (compilers are smart, and would probably calculate it once during building), you can write down the calculation in a comment too.

Reply:
This is how it's designed, we can track it later also(if we really want to fix this one we can have a separate issue).

Reply:
@nixpanic ,I have added the reason above it now. Thanks.